service/src/builder.rs

```diff
@@ -173,6 +177,42 @@ fn is_next(summary: &Summary) -> bool {
         .any(|dependency| dependency.package_name() == NEXT_NAME)
 }
 
+/// Make sure the project is a binary for legacy projects.
+fn ensure_binary(manifest: &Manifest) -> anyhow::Result<()> {
+    if manifest
+        .targets()
+        .iter()
+        .find(|target| target.is_bin())
+        .is_some()
```

Comment:
```suggestion
        .any(|target| target.is_bin())
```
A bit of a nicer way to check it :smile: